crates/oxc_linter/src/rules/eslint/max_len.rs

```diff
@@ -0,0 +1,781 @@
+    fn is_comment(
+        &self,
+        line_index: usize,
+        source_text: &str,
+        comments: &[(CommentKind, Span)],
+        line_span: &Span,
+    ) -> (bool, bool, bool) {
+        let mut has_comment = false;
+        let mut is_trailing = false;
+        let mut is_full_line_comment = false;
+
+        for &(kind, span) in comments.iter() {
+            let comment_start_line = source_text[..span.start as usize].matches('\n').count();
+            let comment_end_line = source_text[..span.end as usize].matches('\n').count();
+
+            match kind {
+                CommentKind::SingleLine if comment_start_line == line_index => {
+                    has_comment = true;
+                    // The start point of a SingleLine comment doesn't include the "//"
+                    // so we need to move 2 characters back when calculating.
+                    is_trailing = span.start - COMMENT_LENGTH > line_span.start
+                        && !source_text
+                            [line_span.start as usize..(span.start - COMMENT_LENGTH) as usize]
+                            .trim()
+                            .is_empty();
+
+                    is_full_line_comment = !is_trailing;
+                }
+                CommentKind::MultiLine
+                    if (comment_start_line <= line_index && comment_end_line >= line_index) =>
+                {
+                    has_comment = true;
+                    if comment_start_line == line_index {
+                        // Check if the multi-line comment ends on the current line
+                        // need to remove the len of "/*"
+                        is_trailing = span.start - COMMENT_LENGTH > line_span.start
+                            && !source_text
+                                [line_span.start as usize..(span.start - COMMENT_LENGTH) as usize]
+                                .trim()
+                                .is_empty();
+
+                        is_full_line_comment = !is_trailing;
+                    }
+
+                    if comment_end_line == line_index {
+                        // the end of a multi-line comment
+                        if line_span.end == span.end + COMMENT_LENGTH {
+                            is_full_line_comment = true;
+                            is_trailing = false;
+                        }
+                    }
+
+                    // the middle of a multi-line comment
+                    if comment_start_line < line_index && comment_end_line > line_index {
+                        is_full_line_comment = true;
+                    }
+                }
+                _ => (),
+            }
+        }
+
+        (has_comment, is_trailing, is_full_line_comment)
+    }
+
+    fn compute_line_length(line: &str, tab_width: usize) -> usize {
+        let mut extra_character_count: isize = 0;
+        for cap in TAB_REGEXP.captures_iter(line) {
```

Comment:
If it still has performance issue, we may need to port it to a non-regexp version, I didn't really understand how this function works in ESLint, so I don't have better suggestion now :(

Reply:
> If it still has performance issue, we may need to port it to a non-regexp version, I didn't really understand how this function works in ESLint, so I don't have better suggestion now :(

now:
```rs
fn compute_line_length(line: &str, tab_width: usize) -> usize {
    let mut extra_character_count: usize = 0;
    let mut last_index: usize = 0; // Record the position after the previous segment

    for segment in line.split('\t').enumerate() {
        let (segment_index, segment) = segment;
        let segment_length = segment.chars().count();

        // Only add to last_index if not on the first segment (since split will not find a tab at the start of the string)
        if segment_index > 0 {
            let total_offset = last_index + extra_character_count;
            let previous_tab_stop_offset =
                if tab_width != 0 { total_offset % tab_width } else { 0 };

            let space_count = tab_width - previous_tab_stop_offset;
            extra_character_count += space_count; // Add the additional space count due to a tab
        }

        // Update last_index to the current position after adding the segment length
        last_index += segment_length;
    }

    // Calculate the full length of the line including extra spaces added due to tabs
    last_index + extra_character_count
}
```